lib/elements/text.js

```diff
@@ -169,12 +170,24 @@ class TextPrompt extends Prompt {
       this.red ? color.red(this.rendered) : this.rendered
     ].join(` `);
 
-    if (this.error) {
-      this.outputError += this.errorMsg.split(`\n`)
-          .reduce((a, l, i) => a + `\n${i ? ' ' : figures.pointerSmall} ${color.red().italic(l)}`, ``);
+    const outputLength = (
+      /* symbol length */
+      2 +
+      /* prompt */
+      this.msg.length +
+      /* delimiter */
+      3 +
+      /* input */
+      this.value.length
+    )
+
+    if (this.error || this.hint) {
+      this.postMessage = (this.error ? this.errorMsg : this.hint).split(`\n`)
+        .reduce((a, l, i) => a + `\n${i ? ' ' : figures.pointerSmall} ${this.error ? color.red().italic(l) : color.gray().italic(l)}`, ``)
+      if (this.hint) this.error = false;
     }
 
-    this.out.write(erase.line + cursor.to(0) + this.outputText + cursor.save + this.outputError + cursor.restore);
+    this.out.write(cursor.to(0) + this.outputText + erase.lineEnd + (this.postMessage ? this.postMessage + cursor.up(lines(this.postMessage - 1)) : '') + cursor.to(outputLength));
```

Comment:
One issue I just thought about with not using `cursor.restore` is that it doesn't account for long messages / input.  If anything wraps, it will cause the message to be truncated, or something similiar.

I will rethink this a little bit.